### PR TITLE
Update: Add difference between slices and choice types

### PR DESCRIPTION
**Pull request by Jasopaum**

### CODE REVIEW

File: src/attribute.ts [hunk 1]
```diff
@@ -175,15 +180,20 @@ export class Attribute {
...
-  addSlice(slice: Attribute) {
-    slice.parent = this.parent
+  addChoice(choice: Attribute) {
+    choice.parent = this.parent
     if (this.isItem) {
-      slice.isItem = true
-      slice.index = this.index
+      choice.isItem = true
+      choice.index = this.index
     }
```
Comment (simonvadee) [inline src/attribute.ts:190]: why did you remove this logic in `addSlice` ?

Comment (simonvadee) [reply src/attribute.ts:190]: also, I don't think a value[x] can be an array so I'd remove this in `addChoice`

Comment (Jasopaum) [reply src/attribute.ts:190]: Because I don't think we need slices to have indices in an array. I saw like that:

parent
  slice1[]
  slice2[]
  slice3[]

Comment (Jasopaum) [reply src/attribute.ts:190]: Hum, it might be true! I kept it because I just replaces what was called slices by choices.
Didn't you have problems without it before?

**Comment by Jasopaum** [comment]
I think spreadTypes is still useful for Pyrog client, isn't it?
